### ISSUE TITLE
`:Gitsigns blame` "Show commit" on same commit hash causes error

### ISSUE THREAD

**Issue by YodaEmbedding** [lewis6991/gitsigns.nvim#1121]
### Description

Running "Show commit" on the same hash causes error.

### Neovim version

v0.10.1

### Operating system and version

macOS 14.6.1

### Expected behavior

Show commit.

### Actual behavior

Results in error:
```
Error executing vim.schedule lua callback: ...cal/share/nvim/lazy/gitsigns.nvim/lua/gitsigns/async.lua:95: The async coroutine failed: ...cal/share/nvim/lazy/gitsigns.nvim/lua/gitsigns/blame.lua:194: Failed to rename buffer
stack traceback:
        [C]: in function 'nvim_buf_set_name'
        ...cal/share/nvim/lazy/gitsigns.nvim/lua/gitsigns/blame.lua:194: in function <...cal/share/nvim/lazy/gitsigns.nvim/lua/gitsigns/blame.lua:188>
stack traceback:
        [C]: in function 'error'
        ...cal/share/nvim/lazy/gitsigns.nvim/lua/gitsigns/async.lua:95: in function 'cb'
        ...cal/share/nvim/lazy/gitsigns.nvim/lua/gitsigns/async.lua:145: in function <...cal/share/nvim/lazy/gitsigns.nvim/lua/gitsigns/async.lua:144>
```

### Minimal config

```Lua
-- Can be provided if maintainer cannot reproduce.
```

### Steps to reproduce

1. `:Gitsigns blame`
2. `s` to "Show commit"
3. Close window (`C-w c`)
4. `s` to "Show commit"

**Comment by mikehaertl** [comment]
I see a similar error on random occasions when I blame from a buffer (so not on a commit or something like OP).

Before I create another issue I'd like to verify that it's really a different problem. My error also includes `async.lua:95` like above.

Here's my error message:

```
Error executing Lua callback: .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/async.lua:95: The async coroutine failed: .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/async.lua:95: The async coroutine faile
d: .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/blame.lua:104: Invalid 'replacement string' item: expected String, got nil
stack traceback:
        [C]: in function 'nvim_buf_set_lines'
        .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/blame.lua:104: in function 'render'
        .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/blame.lua:276: in function <.../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/blame.lua:253>
stack traceback:
        [C]: in function 'error'
        .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/async.lua:95: in function 'step'
        .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/async.lua:130: in function 'f'
        ...al/share/nvim/plugged/gitsigns.nvim/lua/gitsigns/cli.lua:96: in function <...al/share/nvim/plugged/gitsigns.nvim/lua/gitsigns/cli.lua:61>
stack traceback:
        [C]: in function 'error'
        .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/async.lua:95: in function 'step'
        .../share/nvim/plugged/gitsigns.nvim/lua/gitsigns/async.lua:130: in function 'run'
        ....local/share/nvim/plugged/gitsigns.nvim/lua/gitsigns.lua:119: in function <....local/share/nvim/plugged/gitsigns.nvim/lua/gitsigns.lua:118>
```

**Comment by al-eax** [comment]
I got the same error on windows. Same happens when trying to open the Commit with in a new tab `S`.
When pressing `Enter` in the history log before pressing `s` or `S`, nothing happens

**Comment by spaceman1412** [comment]
Me too, i run to the same error

**Comment by antoinemadec** [comment]
@YodaEmbedding @mikehaertl @al-eax @spaceman1412 this is fixed in https://github.com/lewis6991/gitsigns.nvim/pull/1136

Could you please try it on your end and confirm it fixes your issues?

**Comment by al-eax** [comment]
@antoinemadec thank you so much.
I can confirm: the issue is fixed on windows.

**Comment by mikehaertl** [comment]
> @antoinemadec It doesn't seem completely gone for me:

Opened #1139 for this which was fixed by #1140.
